package com.redhat.ceylon.compiler.java.test.statement.trycatch;

class E1 extends .ceylon.language.Exception {
    
    E1(final .ceylon.language.String description, final .java.lang.Throwable cause) {
        super(description, cause);
    }
}
class E2 extends .ceylon.language.Exception {
    
    E2(final .ceylon.language.String description, final .java.lang.Throwable cause) {
        super(description, cause);
    }
}
class TryCatchUnion {
    
    TryCatchUnion() {
    }
    
    private final void t$priv() {
    }
    
    private final void c$priv(final .java.lang.Throwable e) {
    }
    
    private final void m$priv() {
        try {
            t$priv();
        } catch (final .com.redhat.ceylon.compiler.java.test.statement.trycatch.E1 e) {
            c$priv((.java.lang.Throwable)e);
        } catch (final .com.redhat.ceylon.compiler.java.test.statement.trycatch.E2 e) {
            c$priv((.java.lang.Throwable)e);
        }
    }
    
    public static void main(.java.lang.String[] args) {
        .ceylon.language.process_.getProcess$().setupArguments(args);
        new .com.redhat.ceylon.compiler.java.test.statement.trycatch.TryCatchUnion();
    }
}